runtime/help/commands.md

```diff
@@ -42,6 +42,10 @@ quotes here but these are not necessary when entering the command in micro.
 
    Note that `search` must be a valid regex (unless `-l` is passed). If one 
    of the arguments does not have any spaces in it, you may omit the quotes.
+   In case the search is done non-literal (without `-l`) the character `$` can't
+   be simply added, since `Regexp.Expand()` is used in the background to add
+   more complex template replacement functionality. To add a `$` a double `$$`
+   must be used.
```

Comment:
This won't say much to a user who doesn't [speak Go](https://go-language.org/go-docs/regexp/#Regexp.Expand).  How about a summary like:
```suggestion

   In case the search is done non-literal (without `-l`), the 'value'
   is interpreted as a template:
   * `$3` or `${3}` substitutes the submatch of the 3rd (capturing group)
   * `$foo` or `${foo}` substitutes the submatch of the (?P<foo>named group)
   * You have to write `$$` to substitutes a literal dollar.
```
I don't think we need to elaborate on exact lexing e.g. "$10 is equivalent to ${10}, not ${1}0" — if user needs to care about that they're better off just using ${foo}.

(Well, perhaps do mention "as a Go Regexp.Expand template:" — even if one don't speak Go, knowing it's an existing convention can be reassuring vs. "an ad-hoc thing of this editor".)

Reply:
P.S. while we're here, perhaps 'value' is not the best term :thinking: 

I'd actually say it'd be clearest to document 2 separate usage lines, like some man pages do:
```
replace 'regexp' 'template' '-a'?
replace 'literal_text' 'replacement' '-l'
```
but not sure it fits the doc's style, and it's hard to explain how the flags can be combined (is it '-l' '-a'? '-l -a'?  '-la'?)

[Out of scope for this PR: perhaps the literal/regexp modes are sufficiently different that they deserve splitting a separate command, e.g. `replacestring`?  Also provides some discoverability by tab completion? :thinking:]

Reply:
> I'd actually say it'd be clearest to document 2 separate usage lines, like some man pages do:

Unfortunately the behavior isn't that common as expected and I think we should solve that in the first place than precising corner cases within the documentation.

> perhaps the literal/regexp modes are sufficiently different that they deserve splitting a separate command

Many GUI editors treat them in the same way with a search and an replace mask, while you've to set a checkbox for regex capabilities.
I wouldn't separate them, but they need to behave as the user would expect and currently it's everything else than obvious.

My suggestion would be to prevent the usage of `Expand` in a literal search. Just do a plain replace. :thinking: